src/pages/home/sidebar/SidebarScreen/FloatingActionButtonAndPopover.js

```diff
@@ -146,19 +151,17 @@ function FloatingActionButtonAndPopover(props) {
         }
     };
 
-    useEffect(
-        () => {
-            const navigationState = props.navigation.getState();
-            const routes = lodashGet(navigationState, 'routes', []);
-            const currentRoute = routes[navigationState.index];
-            if (currentRoute && ![NAVIGATORS.CENTRAL_PANE_NAVIGATOR, SCREENS.HOME].includes(currentRoute.name)) {
-                return;
-            }
+    useEffect(() => {
+        const navigationState = props.navigation.getState();
+        const routes = lodashGet(navigationState, 'routes', []);
+        const currentRoute = routes[navigationState.index];
+        if (currentRoute && ![NAVIGATORS.CENTRAL_PANE_NAVIGATOR, SCREENS.HOME].includes(currentRoute.name)) {
+            return;
+        }
+        if (!props.showDownloadAppBanner || !Browser.isMobile()) {
             Welcome.show({routes, showCreateMenu});
-        },
-        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Just noticed these few lines getting removed, were they meant to be part of this PR?

Reply:
hmm yeah it was intentional, but I realize we can have this be dependent on when props.showDownloadAppBanner  changes, so I pushed a update to avoid running on every render